        with self.assertRaises(Exception):
            prog.exception('8: Baley Afton')     # exception doesn't occur